packages/js/product-editor/src/utils/handle-confirm.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+export type HandleConfirmProps = {
+	message?: string;
+	onOk(): void;
+	onCancel?(): void;
+};
+
+export async function handleConfirm( {
+	message = __( 'Are you sure?', 'woocommerce' ),
+	onOk,
+	onCancel,
+}: HandleConfirmProps ) {
+	// eslint-disable-next-line no-alert
+	const value = window.confirm( message );
+
+	if ( value === null ) {
+		onCancel?.();
+		return;
+	}
+
+	onOk();
```

Comment:
This does not handle the cancel case, as when cancel is clicked, `false` is returned.

This can be re-written as

```
// eslint-disable-next-line no-alert
	if ( window.confirm( message ) ) {
		onOk?.();
		return;
	}

	onCancel?.();
```

Reply:
@mattsherman Good catch. I've made the change you suggested, although I will note that I'm currently unable to test this due to the issues raised in https://github.com/woocommerce/woocommerce/pull/43441#issuecomment-1985455078